src/main/java/ninja/S3Dispatcher.java

```
@@ -115,6 +117,23 @@
         headerOverrides.put("response-content-encoding", "Content-Encoding");
     }
 
+    private static final Set<String> domains;
+
+    static {
+        domains = new HashSet<>();
+        domains.add(".localhost");
+        domains.add(".127.0.0.1");
+
+        try {
+            InetAddress myself = InetAddress.getLocalHost();
+            domains.add('.' + myself.getHostAddress());
+            domains.add('.' + myself.getHostName());
+            domains.add('.' + myself.getCanonicalHostName());
+        } catch (Exception e) {
+            // reaching this point, we failed to resolve the local host name. tant pis.
```

eitehr Exceptions.handle(e); or if we are 110% sure that this cannot happen or that we truly want to ignore this fact Exceptions.ignore(e);

---

Ignoring it. The exception may be thrown if a machine has problems resolving its own network configuration (i.e. host name), but we cannot do much about it.

---

Exceptions.ignore is the proper way to ignore an exception as it shuts down sonarlint and also provides a way to still enable logging for such cases - which once every 5 years yields an exception which we shoudln't have ignored :-P